src/components/prompt/prompt.jsx

```diff
@@ -0,0 +1,74 @@
+const React = require('react');
+const ReactModal = require('react-modal');
+const bindAll = require('lodash.bindall');
+const Box = require('../box/box.jsx');
+
+const styles = require('./prompt.css');
+
+class PromptComponent extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleOk',
+            'handleKeyPress'
+        ]);
+    }
+    handleKeyPress (event) {
+        if (event.key === 'Enter') this.handleOk();
+    }
+    handleOk () {
+        this.props.onOk(this.input.value);
+    }
+    render () {
+        return (
+            <ReactModal
+                isOpen
+                className={styles.modalContent}
+                overlayClassName={styles.modalOverlay}
+                onRequestClose={this.props.onCancel}
+            >
+                <Box className={styles.body}>
+                    <Box className={styles.label}>
+                        {this.props.label}
+                    </Box>
+                    <Box>
+                        <input
+                            autoFocus
+                            className={styles.input}
+                            placeholder={this.props.placeholder}
+                            ref={el => (this.input = el)}
+                            onKeyPress={this.handleKeyPress}
+                        />
+                    </Box>
+                    <Box className={styles.buttonRow}>
+                        <button
+                            className={styles.cancelButton}
+                            onClick={this.props.onCancel}
+                        >
+                            Cancel
+                        </button>
+                        <button
+                            className={styles.okButton}
+                            onClick={this.handleOk}
+                        >
+                            OK
+                        </button>
+                    </Box>
+                </Box>
+            </ReactModal>
+        );
+    }
+}
+
+PromptComponent.propTypes = {
+    label: React.PropTypes.string.isRequired,
+    onCancel: React.PropTypes.func.isRequired,
+    onOk: React.PropTypes.func.isRequired,
+    placeholder: React.PropTypes.string
+};
+
+PromptComponent.defaultProps = {
+    placeholder: '...'
+};
+
+module.exports = PromptComponent;
```

Comment:
The component should only call props for its handlers, and the container should implement them.

It doesn't matter too much at this point, other than for consistency, but as the components grow, it provides the split between the presentation and business logic.